### ISSUE TITLE
Give better error message when /path/to/workflow.nf is missing 

### ISSUE THREAD

**Issue by alanhoyle** [nextflow-io/nextflow#1951]
## New feature

If there's a spelling error or typo when you try to run a nextflow workflow from a local path, nextflow does not give helpful error messages.  For example, let's imagine the workflow I'm trying to run is /home/user/example_1.nf

```
$ nextflow run ~example_1.nf
N E X T F L O W  ~  version 20.10.0
Not a valid project name: ~example_1.nf

$ nextflow run example1.nf
N E X T F L O W  ~  version 20.10.0
Not a valid project name: example1.nf

$ nextflow run ~/example-1.nf
N E X T F L O W  ~  version 20.10.0
Not a valid project name: /home

$ nextflow run $HOME/example-1.nf
N E X T F L O W  ~  version 20.10.0
Not a valid project name: /home

$ nextflow run ./example-1.nf
N E X T F L O W  ~  version 20.10.0
Can't find git repository config file -- Repository may be corrupted: /home/user/.nextflow/assets/nextflow-io/.

$ nextflow run ../example-1.nf
N E X T F L O W  ~  version 20.10.0
Can't find git repository config file -- Repository may be corrupted: /home/user/.nextflow/assets/nextflow-io/..
```

The most problematic example from above is when it refers to "/home" with nothing else. 

## Usage scenario 

For people who are prone to typos wanting to run nextflow from the command line or in scripts.

## Suggest implementation 

At the very least,  It would be a good deal more friendly if it gave the full path to the attempted script.  Or perhaps a hint like this:  

```
$ nextflow run $HOME/example-1.nf
N E X T F L O W  ~  version 20.10.0
Not a valid project name: /home
  Does '/home/user/example-1.nf' exist?  
```

For the `./example-1.nf` or `../example-1.nf`examples one imagines that it shouldn't be trying to find a git repository named "`.`" or "`..`".  

Note that we already have better behavior if we specify a directory rather than `blah.nf`:

```
$ nextflow run /home/user
N E X T F L O W  ~  version 20.10.0
Missing project main script: /home/user/main.nf
```

**Comment by abhi18av** [comment]
Hi @alanhoyle , thanks for bringing this up!

There has been some [discussion](https://github.com/nextflow-io/nextflow/issues/1820) and [work](https://github.com/nextflow-io/nextflow/pull/1903) previously in this direction.

If the requested feature isn't too different from the one mentioned in #1903 , then shall we move the discussion to that issue?

**Comment by alanhoyle** [comment]
I think this is different.  This needs to be addressed by [... digs into nextflow github  ...]  I think better behavior in the scm/AssetManager.groovy so I'm not sure that it would really fold into the same thing.  

 https://github.com/nextflow-io/nextflow/blob/master/modules/nextflow/src/main/groovy/nextflow/scm/AssetManager.groovy
 
 I note that this misleading `Not a valid project name: /home` error has been around for at least 3 years:
 
 https://groups.google.com/g/nextflow/c/HfzViEXU3oU

**Comment by abhi18av** [comment]
Hi @alanhoyle , 

I think that you are right that the issue outlined here (https://github.com/nextflow-io/nextflow/issues/1951)  and  https://github.com/nextflow-io/nextflow/issues/1820 are somewhat different in scope and PR https://github.com/nextflow-io/nextflow/pull/1903 was initiated with the latter issue in mind.

This use-case is more related to the initial entrypoint of the NF parsing process.

**Comment by pditommaso** [comment]
I've improved a bit this to make the error reporting more consistent  

```
» nextflow run ~example_1.nf
N E X T F L O W  ~  version 21.11.0-edge
Cannot find script file: ~example_1.nf

» nextflow run example_1.nf
N E X T F L O W  ~  version 21.11.0-edge
Cannot find script file: example_1.nf

» nextflow run ~/example_1.nf
N E X T F L O W  ~  version 21.11.0-edge
Cannot find script file: /Users/pditommaso/example_1.nf

» nextflow run $HOME/example_1.nf
N E X T F L O W  ~  version 21.11.0-edge
Cannot find script file: /Users/pditommaso/example_1.nf

» nextflow run ./example_1.nf
N E X T F L O W  ~  version 21.11.0-edge
Cannot find script file: ./example_1.nf

» nextflow run ../example_1.nf
N E X T F L O W  ~  version 21.11.0-edge
Cannot find script file: ../example_1.nf

» nextflow run foo/bar
N E X T F L O W  ~  version 21.11.0-edge
Pulling foo/bar ...
WARN: Cannot read project manifest -- Cause: Remote resource not found: https://api.github.com/repos/foo/bar/contents/nextflow.config
Remote resource not found: https://api.github.com/repos/foo/bar/contents/main.nf

» nextflow run foo
N E X T F L O W  ~  version 21.11.0-edge
Pulling nextflow-io/foo ...
WARN: Cannot read project manifest -- Cause: Remote resource not found: https://api.github.com/repos/nextflow-io/foo/contents/nextflow.config
Remote resource not found: https://api.github.com/repos/nextflow-io/foo/contents/main.nf
```

Solved by e1a0b240a

**Comment by pditommaso** [comment]
Release along with version 21.12.1-edge
